.github/workflows/validate.yml

```diff
@@ -16,16 +16,33 @@ jobs:
       - name: Check out repo
         uses: actions/checkout@v4
 
+      - name: Set up pnpm
+        uses: pnpm/action-setup@v3
+
       - name: Set up Node.js
         uses: actions/setup-node@v4
         with:
           node-version-file: package.json
 
+      - name: Get pnpm store directory
+        id: pnpm-cache
+        shell: bash
+        run: |
+          echo "STORE_PATH=$(pnpm store path)" >> $GITHUB_OUTPUT
+
+      - name: Cache node modules
+        uses: actions/cache@v4
+        with:
+          path: ${{ steps.pnpm-cache.outputs.STORE_PATH }}
+          key: ${{ runner.os }}-pnpm-store-${{ hashFiles('**/pnpm-lock.yaml') }}
+          restore-keys: |
+            ${{ runner.os }}-pnpm-store-
+
```

Comment:
Probably shouldn't have pointed to playroom as a reference for a PNPM package as it's CI is a bit outdated. This cache stuff isn't necessary as the `setup-node` action can cache node modules for you with a single option: `cache: pnpm`.

See https://github.com/seek-oss/eslint-config-seek/blob/98bfbf88284b962d5720f4199c1c32c7060c5bc7/.github/workflows/test.yml#L24.